vm/systemSmartContracts/delegation.go

```diff
@@ -1280,7 +1270,11 @@ func getTransferBackFromVMOutput(vmOutput *vmcommon.VMOutput) *big.Int {
 	return transferBack
 }
 
-func moveNodeFromList(sndList []*NodesData, dstList []*NodesData, key []byte) {
+func moveNodeFromList(
+	sndList []*NodesData,
+	dstList []*NodesData,
+	key []byte,
+) ([]*NodesData, []*NodesData) {
 	for i, nodeData := range sndList {
 		if bytes.Equal(nodeData.BLSKey, key) {
 			copy(sndList[i:], sndList[i+1:])
```

Comment:
dstList is not used in this method and this copy should be refactored like you have done with the other ones as right now it not works as expected

Reply:
dstList is used. as it is appended.